tests/test_pipelines_image_classification.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2021 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+from tests.test_pipelines_common import CustomInputPipelineCommonMixin
+from transformers import Pipeline, is_torch_available, is_vision_available
+from transformers.pipelines import pipeline
+from transformers.testing_utils import _run_slow_tests, require_torch, require_vision
+
+
+if is_vision_available():
+    from PIL import Image
+else:
+
+    class Image:
+        @staticmethod
+        def open(*args, **kwargs):
+            pass
```

Comment:
I wonder whether it might be better to provide a warning here so that one knows it's a dummy class and wonders what happened to the `PIL.Image.open()`

Reply:
This is a dummy class that's only useful for testing purposes (it's in a test class). The class property `valid_input` is called when the class is created, therefore running the following code: `Image.open("tests/fixtures/coco.jpg")`. It does so even when PIL isn't installed, which would then crash were a suite run without `PIL` installed.